src/test/java/com/google/devtools/build/lib/rules/cpp/ShowIncludesFilterTest.java

```diff
@@ -93,17 +93,31 @@ public void testMatchAllOfNotePrefix() throws IOException {
   }
 
   @Test
-  public void testMatchAllOfNotePrefixWithAbsolutePath() throws IOException {
+  public void testFindHeaderFromAbsolutePathUnderExecroot() throws IOException {
     // "Note: including file:" is the prefix
     filterOutputStream.write(
-        getBytes("Note: including file: C:\\tmp\\xxxx\\execroot\\__main__\\bar.h"));
+        getBytes("Note: including file: C:\\tmp\\xxxx\\execroot\\__main__\\foo\\bar\\bar.h"));
     filterOutputStream.flush();
     // flush to output should not work, waiting for newline
     assertThat(output.toString()).isEmpty();
     filterOutputStream.write(getBytes("\n"));
     // It's a match, output should be filtered, dependency on bar.h should be found.
     assertThat(output.toString()).isEmpty();
-    assertThat(showIncludesFilter.getDependencies()).contains("bar.h");
+    assertThat(showIncludesFilter.getDependencies()).contains("foo\\bar\\bar.h");
+  }
+
+  @Test
+  public void testFindHeaderFromAbsolutePathOutsideExecroot() throws IOException {
```

Comment:
In a comment, please reference the bug for which this is a regression test.

Reply:
Thanks, done!